actions/docs-verifier/src/ActionRunner/Program.cs

```diff
@@ -68,9 +68,10 @@
             file = file.Remove(linkError.UrlSpan.Start, linkError.UrlSpan.Length);
 
             string newLink = Path.GetRelativePath(Path.GetDirectoryName(linkError.File)!, newAbsolutePath);
-            if (linkError.Link.Contains('#', StringComparison.Ordinal))
+            int queryOrHeadingIndex = Math.Min(linkError.Link.LastIndexOf('#'), linkError.Link.LastIndexOf('?'));
+            if (queryOrHeadingIndex > -1)
             {
-                newLink += newLink.Substring(linkError.Link.IndexOf('#', StringComparison.Ordinal));
+                newLink += linkError.Link.Substring(queryOrHeadingIndex);
```

Comment:
This was a bug in the previous PR.

I'll see if I can setup a reasonable test infrastructure for tryfix feature.